docs/api-spec.md

```diff
@@ -0,0 +1,322 @@
+# Tekton Pipelines API Specification
+
+<!-- toc -->
+- [Abstract](#abstract)
+- [Background](#background)
+- [Modifying This Specification](#modifying-this-specification)
+- [Resource Overview](#resource-overview)
+  * [`TaskRun`](#-taskrun-)
+- [Detailed Resources - v1beta1](#detailed-resources---v1beta1)
+  * [`TaskRun`](#-taskrun--1)
+    + [Metadata](#metadata)
+    + [Spec](#spec)
+    + [Status](#status)
+- [Status Signalling](#status-signalling)
+- [Listing Resources](#listing-resources)
+- [Detailed Resource Types - v1beta1](#detailed-resource-types---v1beta1)
+  * [`ArrayOrString`](#arrayorstring)
+  * [`ContainerStateRunning`](#containerstaterunning)
+  * [`ContainerStateWaiting`](#containerstatewaiting)
+  * [`ContainerStateTerminated`](#containerstateterminated)
+  * [`EnvVar`](#envvar)
+  * [`Param`](#param)
+  * [`ParamSpec`](#paramspec)
+  * [`Step`](#step)
+  * [`StepState`](#stepstate)
+  * [`TaskResult`](#taskresult)
+  * [`TaskRunResult`](#taskrunresult)
+  * [`TaskSpec`](#taskspec)
+  * [`WorkspaceBinding`](#workspacebinding)
+  * [`WorkspaceDeclaration`](#workspacedeclaration)
+<!-- /toc -->
+
+## Abstract
+
+The Tekton Pipelines platform provides common abstractions for describing and executing container-based, run-to-completion workflows, typipcally in service of CI/CD scenarios. This document describes the structure, lifecycle and management of Knative resources in the context of the [Kubernetes Resource Model](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture/resource-management.md). An understanding of the Kubernetes API interface and the capabilities of [Kubernetes Custom Resources](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) is assumed.
+
+This document does not define the [runtime contract](https://tekton.dev/docs/pipelines/container-contract/) nor prescribe specific implementations of supporting services such as access control, observability, or resource management.
```

Comment:
I don't mind it being in a separate document. It's not currently phrased in terms of "platform implementations must satisfy this contract" so much as "users should expect this contract", but maybe that's just fine since the vast majority of readers will be users, and very few will be prospective platform providers.

In the meantime it's probably enough to just say "this isn't the runtime contract, that's over there" and when/if we get any confusion we can resolve it. Launch and iterate 😄